gitops-v2/validate/main.yaml

```diff
@@ -0,0 +1,48 @@
+parameters:
+  - name: poolVmImage
+    type: string
+    default: "ubuntu-18.04"
+  - name: poolName
+    type: string
+    default: ""
+  - name: binaries
+    type: object
+    default:
+      kustomize:
+        tag: "v4.5.1"
+        sha: "b8e16b303d9aa210d135710efb63f39ec15ae2a5fb5dedf38698f512f95cd2bd"
+
+stages:
+  - stage: validate
+    jobs:
+      - job: kustomize
+        pool:
+          ${{ if not(eq(parameters.poolName, '')) }}:
+            name: ${{ parameters.poolName }}
+          ${{ if not(eq(parameters.poolVmImage, '')) }}:
+            vmImage: ${{ parameters.poolVmImage }}
+        continueOnError: false
+        steps:
+          - bash: |
+              set -e
+              wget https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2F${KUSTOMIZE_TAG}/kustomize_${KUSTOMIZE_TAG}_linux_amd64.tar.gz
+              tar -xvf kustomize_v4.5.1_linux_amd64.tar.gz
+              DOWNLOAD_KUSTOMIZE_SHA=$(openssl sha1 -sha256 kustomize | awk '{print $2}')
+              if [[ "${KUSTOMIZE_SHA}" != "${DOWNLOAD_KUSTOMIZE_SHA}" ]]; then
+                  echo "Downloaded checksum (${DOWNLOAD_KUSTOMIZE_SHA}) for 'kustomize' does not match expected value: ${KUSTOMIZE_SHA}"
+                  exit 1
+              fi
+              # Go to all folders containing a kustomization.yaml file and run kustomize build
+              while IFS= read -r -d '' file; do
+                  dir=$(echo "$file" | sed -r 's|/[^/]+$||')
+                  cd "$dir"
+                  echo -e "\n======================================================================================" 1>&2
+                  echo "Running  'kustomize build .' in '$dir':" 1>&2
+                  echo "======================================================================================" 1>&2
+                  kustomize build .
+                  cd -
```

Comment:
Does kustomize actually care about cwd? Its `DIR` argument seems to be sufficient? I.e. `kustomize build "$dir"`?

Reply:
Unfortunately not since you do relative file references in kustomize

Reply:
I may not understand what you mean. This works for me:
```
$ cat apps/dev/kustomization.yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization
namespace: lab
resources:
  - configmap.yaml
$ cat apps/dev/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: foo
data:
  foo: bar
$ kustomize build ./apps/dev/
apiVersion: v1
data:
  foo: bar
kind: ConfigMap
metadata:
  name: foo
  namespace: lab
```
My understanding is that resource references in a Kustomization is relative to that file, not to cwd?

Reply:
Ok, my mistake then. I have seen problems with that in the past bu maybe not valid any more.